packages/picasso.js/src/core/chart/index.js

```diff
@@ -426,6 +426,12 @@ function chartFn(definition, context) {
     // Let the "components" array determine order of components
     currentComponents = components.map((comp) => {
       const idx = findComponentIndexByKey(comp.key);
+
+      // Component should not be updated
+      if (excludeFromUpdate.length && excludeFromUpdate.indexOf(comp.key) > -1) {
```

Comment:
Is it intentional that you can exclude a component that is added the same update call? Based on your PR description, the logic doesn't seem to match.

Reply:
It shouldn't be a problem since the component will be filtered out further down anyway.